### PR TITLE
improve PATH handling

### PR DESCRIPTION
**Pull request by astj**
- Make PATH not to end with `:`
- Consider path containing `;` in Windows

This is inspired from #499 , but not depending on external libraries.

### CODE REVIEW

**Comment by itchyny** [comment]
LGTM

**Comment by mattn** [comment]
I suggest to add `/usr/local/go/bin`

File: init_windows.go [hunk 1]
```diff
@@ -5,11 +5,19 @@ package main
...
 	"path/filepath"
+	"strings"
 )
 
 func init() {
 	if p, err := os.Executable(); err == nil {
-		os.Setenv("PATH", filepath.Dir(p)+
-			string(filepath.ListSeparator)+os.Getenv("PATH"))
+		dir := filepath.Dir(p)
+		if strings.Index(dir, ";") > -1 {
```
Comment (Songmu) [inline init_windows.go]: [nit] How about using `strings.ContainsRune`?

File: init_unix.go [hunk 2]
```diff
@@ -5,7 +5,11 @@ package main
 import "os"
 
 func init() {
```
Comment (Songmu) [inline init_unix.go]: How about the following implementation? It is matter of preference so, you do not necessarily have to apply it.

```go
paths := []string{"/sbin", "/usr/sbin", "/usr/bin"}
if p := os.Getenv("PATH"); p != "" {
   paths = append(paths, p)
}
os.Setenv("PATH", strings.Join(paths, string(filepath.ListSeparator)))
```

**Comment by mattn** [comment]
> I suggest to add /usr/local/go/bin

This is suggestion. So I don't mind even if you don't want my idea. Most user probably install Go into /usr/local/go (if it is not a packages). And this is idea to enable `go run` or `go env` without absolute PATHs.

**Comment by astj** [comment]
@Songmu @mattn Thank you for suggestions. But I'd like to keep them unchanged.

**Pull request merged**